module/vision/GoalDetector/src/GoalDetector.cpp

```diff
@@ -591,9 +593,10 @@ namespace vision {
                         Eigen::Matrix3d rGCc_cov = convert<double, 3, 3>(arma::diagmat(
                             VECTOR3_COVARIANCE % covariance_amplifier));  // arma::diagmat(arma::vec3{0.01,0.01,0.001})
 
-
-                        it->measurement.push_back(
-                            Goal::Measurement(Goal::MeasurementType::CENTRE, rGCc_sphr, rGCc_cov));
+                        if (!isinf(rGCc_sphr[0]) && !isinf(rGCc_sphr[1]) && !isinf(rGCc_sphr[2])) {
```

Comment:
`std::isinf`
If we're doing this kind of filtering, it might be good to check `std::isnan` too, also

Reply:
or maybe use `std::isfinite` instead

Reply:
yes, std::isfinite is a much better option